apps/console/src/features/tenants/components/dropdown/tenant-dropdown.tsx

```diff
@@ -0,0 +1,651 @@
+    const [ tempTenantAssociationsList, setTempTenantAssociationsList ] = useState<string[]>(undefined);
+    const [ showTenantAddModal, setShowTenantAddModal ] = useState<boolean>(false);
+    const [ isSwitchTenantsSelected, setIsSwitchTenantsSelected ] = useState<boolean>(false);
+    const [ isSetDefaultTenantInProgress, setIsSetDefaultTenantInProgress ] = useState<boolean>(false);
+    const [ associatedTenants, setAssociatedTenants ] = useState<string[]>([]);
+    const [ defaultTenant, setDefaultTenant ] = useState<string>("");
+    const [ isDropDownOpen, setIsDropDownOpen ] = useState<boolean>(false);
+    const [ organizationId, setOrganizationId ] = useState<string>("");
+    const [ organizationName, setOrganizationName ] = useState<string>("");
+    const [ isCopying, setIsCopying ] = useState<boolean>(false);
+
+    const orgType: OrganizationType = useGetOrganizationType();
+
+    const saasFeatureStatus : FeatureStatus = useCheckFeatureStatus(FeatureGateConstants.SAAS_FEATURES_IDENTIFIER);
+
+    const isSubOrg: boolean = window[ "AppUtils" ].getConfig().organizationName;
+
+    useEffect(() => {
+        if (!isPrivilegedUser && saasFeatureStatus !== FeatureStatus.DISABLED) {
+            getAssociatedTenants()
+                .then((response: TenantRequestResponse) => {
+                    let defaultDomain: string = "";
+                    const tenants: string[] = [];
+
+                    response.associatedTenants.forEach((tenant: TenantInfo) => {
+                        if (tenant.default) {
+                            defaultDomain = tenant.domain;
+                        }
+
+                        tenants.push(tenant.domain);
+                    });
+
+                    dispatch(setTenants<TenantInfo>(response.associatedTenants));
+                    setAssociatedTenants(tenants);
+                    setDefaultTenant(defaultDomain);
+                })
+                .catch((error: any) => {
+                    dispatch(
+                        addAlert({
+                            description:
+                                error?.description &&
+                                t("extensions:manage.features.tenant.notifications." + "getTenants.description"),
+                            level: AlertLevels.ERROR,
+                            message:
+                                error?.description &&
+                                t("extensions:manage.features.tenant.notifications." + "getTenants.message")
+                        })
+                    );
+                });
+        }
+        getOrganizationData();
+    }, []);
+
+    useEffect(() => {
+
+        const association: TenantAssociationsInterface = {
+            associatedTenants: associatedTenants,
+            currentTenant: tenantDomain,
+            defaultTenant: defaultTenant,
+            username: email ? email : username
```

Comment:
Shall we shorten this as follows,
```suggestion
            username: email ?? username
```